src/polycap-description.c

```diff
@@ -94,10 +113,161 @@ polycap_description* polycap_description_new_from_file(const char *filename)
+	description->nelem = nelem;
+	description->density = density;
+	for(i=0; i<description->nelem; i++){
+		description->iz[i] = iz[i];
+		description->wi[i] = wi[i]; //assumes weights are already provided as fractions (not percentages)
+	}
+
+	//allocate profile memory
+	description->profile = malloc(sizeof(polycap_profile));
+	if(description->profile == NULL){
+		printf("Could not allocate description->profile memory.\n");
+		exit(1);
+	}
+	description->profile->z = malloc(sizeof(double)*(profile->nmax+1));
+	if(description->profile->z == NULL){
+		printf("Could not allocate description->profile->z memory.\n");
+		exit(1);
+	}
+	description->profile->cap = malloc(sizeof(double)*(profile->nmax+1));
+	if(description->profile->cap == NULL){
+		printf("Could not allocate description->profile->cap memory.\n");
+		exit(1);
+	}
+	description->profile->ext = malloc(sizeof(double)*(profile->nmax+1));
+	if(description->profile->ext == NULL){
+		printf("Could not allocate description->profile->ext memory.\n");
+		exit(1);
+	}
+	
+	// copy description->profile values to profile
+	for(i=0;i<=profile->nmax;i++){
+		description->profile->z[i] = profile->z[i];
+		description->profile->cap[i] = profile->cap[i];
+		description->profile->ext[i] = profile->ext[i];
+	}
+
+	return description;
+}
+
+//===========================================
+// get the polycap_profile from a polycap_description
+polycap_profile* polycap_description_get_profile(polycap_description *description)
+{
+	polycap_profile *profile;
+	int i;
+
+	//allocate profile memory
+	profile = malloc(sizeof(polycap_profile));
+	if(profile == NULL){
+		printf("Could not allocate profile memory.\n");
+		exit(1);
+	}
+	profile->nmax = description->profile->nmax;
+	profile->z = malloc(sizeof(double)*(profile->nmax+1));
+	if(profile->z == NULL){
+		printf("Could not allocate profile->z memory.\n");
+		exit(1);
+	}
+	profile->cap = malloc(sizeof(double)*(profile->nmax+1));
+	if(profile->cap == NULL){
+		printf("Could not allocate profile->cap memory.\n");
+		exit(1);
+	}
+	profile->ext = malloc(sizeof(double)*(profile->nmax+1));
+	if(profile->ext == NULL){
+		printf("Could not allocate profile->ext memory.\n");
+		exit(1);
+	}
+
+	// copy description->profile values to profile
+	for(i=0;i<=profile->nmax;i++){
+		profile->z[i] = description->profile->z[i];
+		profile->cap[i] = description->profile->cap[i];
+		profile->ext[i] = description->profile->ext[i];
+	}
+
+	return profile;
+
+}
+
+//===========================================
+// for a given array of energies, and a full polycap_description, get the transmission efficiencies.
+int polycap_description_get_transmission_efficiencies(polycap_description *description, size_t n_energies, double *energies, double **efficiencies)
+{
+//why return an integer? Something like 0 for succes, -1 for fail, ...?
+
+//this is essentially the full old polycap program. Write later, as it may use some additional functions such as polycap_photon_launch.
```

Comment:
Yep de bedoeling is dat je idd de volledige polycap gebruikt maar dan door intern de nieuwe API te gebruiken.